extensions/amp-animation/0.1/keyframes-extractor.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @extends {CSSRule}
+ * @see http://www.w3.org/TR/DOM-Level-2-Style/css.html#CSS-CSSKeyframesRule
+ */
+class CSSKeyframesRuleDef {
+  constructor() {
+    /** @type {string} */
+    this.name;
```

Comment:
I tried, but unfortunately can't. The underlying `CSSRule` is declared as a class, not interface. See [this](https://github.com/google/closure-compiler/blob/311397bdfd1a0183fd6984a1522205ea426296f5/externs/browser/w3c_css.js#L205). This is kind of what our `Def` suffix means at this point. I'm going to contribute the same declarations to them, but we are slightly behind their externs version.